src/app/api/Mock/MockApi.ts

```diff
@@ -241,6 +271,26 @@ export default class MockClient implements API {
         return sheriffShifts;
     }
 
+    async updateSelectedShifts(shiftIds: IdType[], shiftUpdates: ShiftUpdates): Promise<Shift[]> {
+        await randomDelay();
+        
+        if (!shiftIds) {
+            throw new Error('No ID specified');
+        }
+
+        let updatedShifts: Shift[] = [];
+
+        shiftIds.forEach(shiftId => {
+            const shiftIndex = sheriffShifts.findIndex((shift) => shift.id === shiftId);
+            if (shiftIndex < 0) {
+                throw Error(`No shift could be located for ${shiftId}`);
+            }
+            updatedShifts.push(this.modifyShift(sheriffShifts[shiftIndex], shiftUpdates));
```

Comment:
*EDIT* Realized that you actually DO want to modify the records in memory, but leaving the comment below as just a gotcha in the event you hadn't considered it.  But it looks like you may have already considered it.

mock api so not a big deal, but you probably want to spread the shift into the `modifyShift` function to avoid weird memory/reference errors.  I would also probably use just find here as it will return you either the record or undefined since you don't really use the index later on.
```ts
const foundShift = sheriffShifts.find(s=>s.id==shiftId);
if(foundShift){
  updatedShifts.push(this.modifyShift({...foundShift},shiftUpdates))
```